pkg/sources/gitlab/gitlab.go

```diff
@@ -386,3 +405,18 @@ func (s *Source) basicAuthSuccessful(apiClient *gitlab.Client) bool {
 	}
 	return false
 }
+
+// setProgressCompleteWithRepo calls the s.SetProgressComplete after safely setting up the encoded resume info string.
+func (s *Source) setProgressCompleteWithRepo(index int, repoURL string) {
+	s.resumeInfoMutex.Lock()
+	defer s.resumeInfoMutex.Unlock()
+
+	// Add the repoURL to the resume info slice.
+	s.resumeInfoSlice = append(s.resumeInfoSlice, repoURL)
+	sort.Strings(s.resumeInfoSlice)
```

Comment:
**Nit (optional):** We could more efficiently insert `repoURL` into the slice since it should already be sorted, though I kind of like the stability of sorting every time.

Reply:
True. Maybe something to keep an eye on, as though I don't expect anyone to have like 10k+ repos to sort nor the concurrency to have that many listed in the resume info, you never know.